packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -2669,39 +2718,30 @@ export class ProjectService extends BaseService {
                 const uniqueSpaceUuids = [
                     ...new Set(savedCharts.map((chart) => chart.spaceUuid)),
                 ];
-                const exploreCacheKeys: Record<string, boolean> = {};
-                const exploreCache: Record<string, Explore> = {};
-
-                const explorePromises = savedCharts.reduce<
-                    Promise<{ key: string; explore: Explore }>[]
-                >((acc, chart) => {
-                    const key = chart.tableName;
-                    if (!exploreCacheKeys[key]) {
-                        acc.push(
-                            this.getExplore(user, chart.projectUuid, key).then(
-                                (explore) => ({ key, explore }),
-                            ),
-                        );
-                        exploreCacheKeys[key] = true;
-                    }
-                    return acc;
-                }, []);
-
-                const [spaceAccessMap, resolvedExplores] = await Promise.all([
-                    this.spaceModel.getSpacesForAccessCheck(uniqueSpaceUuids),
-                    Promise.all(explorePromises),
-                ]);
-                const userSpacesAccess =
-                    await this.spaceModel.getUserSpacesAccess(
-                        user.userUuid,
-                        uniqueSpaceUuids,
-                    );
 
-                resolvedExplores.forEach(({ key, explore }) => {
-                    exploreCache[key] = explore;
-                });
+                if (savedCharts.length === 0) {
+                    return [];
+                }
 
-                const filterPromises = savedCharts.map(async (savedChart) => {
+                const [spaceAccessMap, exploresMap, userSpacesAccess] =
+                    await Promise.all([
+                        this.spaceModel.getSpacesForAccessCheck(
+                            uniqueSpaceUuids,
+                        ),
+                        this.findExplores({
+                            user,
+                            projectUuid: savedCharts[0].projectUuid, // TODO: route should be updated to be project/dashboard specific. For now we pick it from first chart as they all should be from the same project
+                            exploreNames: savedCharts.map(
+                                (chart) => chart.tableName,
+                            ),
```

Comment:
can we pass `organizationUuid` from the `user.organizationUuid` here?

Reply:
there is no `organizationUuid` in the code

Reply:
ah, got it 👍 thanks, adding it now